metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,299 @@
+#include "tax_classifier.hpp"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "common/unix_tools.hpp"
+#include "common/utils/string_utils.hpp"
+
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+bool TaxonomyBase::assign_label_type(const std::string &sample_label) {
+    if (utils::starts_with(sample_label, "gi|")) {
+        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
+        label_type = GEN_BANK;
+        return true;
+    } else if (utils::starts_with(utils::split_string(sample_label, ":")[1], "taxid|")) {
+        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
+        label_type = TAXID;
+        return false;
+    }
+
+    logger->error("Can't determine the type of the given label {}. "
+                  "Make sure the labels are in a recognized format.", sample_label);
+    exit(1);
+}
```

Comment:
it's quite unclear what this function is doing. Also, it's called only in one place, so I'd recommend removing it and move the code right there.

Or, if you need to call this function also in child classes, make it a properly: return label_type -- and try/false can be checked instantly after the call, e.g.:
```c++
... label_type = get_label_type(sample_label);

bool require_accversion_to_taxid_map = (label_type == GEN_BANK);
```

Reply:
I have moved it directly to the constructor. Thanks for the suggestion!